silver/tests/spec/test_payment_method.py

```diff
@@ -1,27 +1,308 @@
+            PaymentMethod.States.Enabled})
+
+        processor_url = reverse('payment-processor-detail', kwargs={
+            'processor_name': 'manual'
+        })
+
+        url = reverse('payment-method-list', kwargs={
+            'customer_pk': self.customer.pk
+        })
+
+        for state in invalid_initial_states:
+            response = self.client.post(url, data={
+                'payment_processor': processor_url,
+                'state': state
+            }, format='json')
+
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_put_detail_state_transitions(self):
+        states = PaymentMethod.States.as_list()
+        permutations = [(old, new) for old in states for new in states]
+
+        valid_transitions = set()
+        for _, transition in iteritems(PaymentMethod.state_transitions):
+            if isinstance(transition['source'], (tuple, list)):
+                unmerged_transitions = set([(old, transition['target']) for
+                                            old in transition['source']])
+            else:
+                unmerged_transitions = {(transition['source'],
+                                         transition['target'])}
+
+            valid_transitions = valid_transitions.union(unmerged_transitions)
+
+        # add x -> x transitions as they are allowed
+        valid_transitions = valid_transitions.union(
+            set([(x, x) for x in states]))
+
+        valid_transition_list = list(valid_transitions)
+
+        for old, new in permutations:
+            payment_method = self.create_payment_method(customer=self.customer,
+                                                        state=old)
+
+            url = reverse('payment-method-detail', kwargs={
+                'customer_pk': self.customer.pk,
+                'payment_method_id': payment_method.pk
+            })
+            response = self.client.get(url, format='json')
+
+            data = response.data
+            data['state'] = new
+
+            response = self.client.put(url, data=data, format='json')
+
+            message = '{} -> {}: code got {{}} expected {{}}'.format(old, new)
+            if (old, new) in valid_transition_list:
+                self.assertEqual(response.status_code,
+                                 status.HTTP_200_OK,
+                                 message.format(response.status_code,
+                                                status.HTTP_200_OK))
+                self.assertEqual(data, response.data,
+                                 message.format(response.status_code,
+                                                status.HTTP_200_OK))
+            else:
+                self.assertEqual(response.status_code,
+                                 status.HTTP_409_CONFLICT,
+                                 message.format(response.status_code,
+                                                status.HTTP_409_CONFLICT))
```

Comment:
There is no point for this as of yet as the code that should generate the lattice of transitions should still be there, we'll only loose time but not gain much in return (maybe more verbosity for successfully ran tests, not worth it imho).